### PR TITLE
Week 06 day 04 drill

### PR DESCRIPTION
**Pull request by ben-manning**
I feel good with this drill.  It's the one that WDI-22-23 used and it seems pretty in-depth and well explained.

### CODE REVIEW

**Comment by eerwitt** [comment]
Do you think it'd be easier to make a separate repo for the drill starter code?

@cameron @ben.manning I need to do something similar in the regex drill, would it be better to put files in my drill directory or use a separate repo?

**Comment by eerwitt** [comment]
:+1: it's a solid drill, I'd enjoy it.

**Comment by ben-manning** [comment]
@eerwitt I think stylistically we are keeping it in the same spot as the drills.  Does it seem like it's a bit cluttered if that's the case?  Also, if I keep it there does it make sense to walk them through that set up before working on the drill?

**Comment by eerwitt** [comment]
I'm not sure, I can see it as useful either way.

**Comment by ben-manning** [comment]
I'm cool either way.  We're both not really helping are we?

**Comment by eerwitt** [comment]
I believe we've at least established that we're at a stalemate of uncertainty. We may need an outside observer in order for us to come to a decision.

**Comment by ben-manning** [comment]
Lol.  Agreed.

**Comment by cameronjacoby** [comment]
@ben-manning I would put the starter code with tests in a separate repo for sure. Anything you want them to clone should live in a separate repo. Otherwise looks good!

File: week-06-intro-ruby-on-rails/day-04/drill/solution.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        # so the until loop starts again
+      end
+    end
+  end
+  # return the sorted array
+  array
+end
+
+
+# def bubble_sort(array)
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-04/drill/solution.rb]: @ben-manning Is there a reason why this one is commented out? If it's an alternate solution, could you indicate that?

**Comment by ben-manning** [comment]
@cameronjacoby Hey, moved the code to a separate repo and uncommented the solution things.  I think it's good to go.

File: week-06-intro-ruby-on-rails/day-04/drill/readme.md [hunk 2]
```diff
@@ -1 +1,111 @@
 # <img src="https://cloud.githubusercontent.com/assets/7833470/10899314/63829980-8188-11e5-8cdd-4ded5bcb6e36.png" height="60"> Ruby Bubble Sort
+
+Run the tests in the `spec` folder, and write your code in `bubble_sort.rb`.
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-04/drill/readme.md]: @ben-manning Can you move this whole section about running tests/exploring code down to the bottom where you intro the challenge? It doesn't make sense here b/c they haven't cloned anything to get started yet.

**Comment by cameronjacoby** [comment]
@ben-manning Awesome, thanks! One line comment above, and can you also add a solution? You can make a `solution` branch on the starter code repo, then create a `solutions.md` file directly in this drill folder, and link out to the solution branch of the other repo.

**Comment by ben-manning** [comment]
@cameronjacoby Hey, I've got the solutions moved to a different branch on the other repo.  I think the order of this lecture makes a little more sense.  Let me know what you think.

**Comment by cameronjacoby** [comment]
@ben-manning Perfect, thanks! Can you still create a `solutions.md` in this directory and just put a link to the solution branch of the other repo? Just so it's super clear where they can find it.

**Comment by ben-manning** [comment]
@cameronjacoby Cool.  I've got them both in.

**Comment by cameronjacoby** [comment]
@ben-manning Sorry, I meant just a link to the solution branch in `solutions.md` so you don't have to paste the whole solution in.

That wasn't very clear - don't worry, I just added it in!

Everything looks good! :+1:

**Pull request merged**